MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -7906,79 +7910,86 @@ public void initTimeInService() {
         }
     }
 
-    public void initAtB() {
-        retirementDefectionTracker.setLastRetirementRoll(calendar);
-
-        /*
-         * Switch all contracts to AtBContract's
-         */
-        for (Map.Entry<Integer, Mission> me : missions.entrySet()) {
-            Mission m = me.getValue();
-            if (m instanceof Contract && !(m instanceof AtBContract)) {
-                me.setValue(new AtBContract((Contract)m, this));
-            }
-        }
+    public void initAtB(boolean newCampaign) {
+        getRetirementDefectionTracker().setLastRetirementRoll(getCalendar());
 
-        /*
-         * Go through all the personnel records and assume the earliest date is the date
-         * the unit was founded.
-         */
-        Date founding = null;
-        for (Person p : getPersonnel()) {
-            for (LogEntry e : p.getPersonnelLog()) {
-                if (null == founding || e.getDate().before(founding)) {
-                    founding = e.getDate();
+        if (!newCampaign) {
+            /*
+            * Switch all contracts to AtBContract's
+            */
+            for (Map.Entry<Integer, Mission> me : missions.entrySet()) {
```

Comment:
I'm not sure that this approach is correct, as a user may have a mixture of AtB and non-AtB missions.

Reply:
That was the original setup for when a user switched to AtB. I can revise that so it doesn't if you think that's best.